clustering/interface.py

```diff
@@ -0,0 +1,89 @@
+import copy
+import traceback, logging
+from django.conf import settings
+from django.core.cache import cache
+from utils.encryption import create_hash
+from utils.search.search_general import search_prepare_query, perform_solr_query, \
+    search_prepare_parameters
+
+from tasks import cluster_sounds
+from clustering_settings import MAX_RESULTS_FOR_CLUSTERING, CLUSTERING_CACHE_TIME, DEFAULT_FEATURES
+from . import CLUSTERING_RESULT_STATUS_PENDING, CLUSTERING_RESULT_STATUS_FAILED
+
+
+def get_sound_ids_from_solr_query(query_params):
+    """Performs Solr query and returns results as a list of sound ids.
+
+    This method performs a single query to Solr with a very big page size argument so all results are 
+    returned at once. A very big page size will make the clustering take a lot of time to be performed.
+    The number of results to retrieve is defined in the clustering settings file as MAX_RESULTS_FOR_CLUSTERING.
+
```

Comment:
There is not a maximum number of results that SOLR can return. 
It seems that it is more a question of how feasible it is to transmitted the data.

Since I am only interested in retrieving the sound ids, I could set `include_facets` to False to reduce the amount of data that Solr returns.